enhancements/csvless-bundles.md

```diff
@@ -24,209 +24,488 @@ status: implementable
+2 directories, 4 files
+```
 
-Another risk is around operatorhub.io - it will need to be adjusted to support the `registry+v2` bundles. This work is not scoped as part of this enhancement and is done by a seperate team - how much effort is involved there is unclear.
+_Ex. Initialize defaults for `registry+v2` in the current empty directory and generate a dockerfile_
+
+```sh
+$ tree .
+.
+
+0 directories, 0 files
+$ opm bundle init --mediatype 'registry+v2' -d .
+# <logs>
+$ tree .
+.
+├── Dockerfile
+├── manifests
+│   └── deployment.yaml
+└── metadata
+    ├── bundle.yaml
+    ├── annotations.yaml
+    └── dependencies.yaml
+
+2 directories, 5 files
+```
 
-Security impacts are minimal because the number of kubernetes objects that OLM supports is still the same - just the delivery mechanism (the bundle format) is changing.
+_Ex. Fail to detect mediatype on empty directory_
 
-## Design Details
+```sh
+$ tree .
+.
 
-### Implementation Details
+0 directories, 0 files
+$ opm bundle init .
+# <log: "unable to infer mediatype when no manifests are present">
+```
+
+_Ex. Infer mediatype from src manifests_
+
+```sh
+$ tree kiali-csvless
+kiali-csvless
+├── crds.yaml
+├── deployment.yaml
+└── rbac.yaml
+$ opm bundle init -o kiali kiali-csvless
+# <logs>
+$ tree kiali
+kiali
+├── manifests
+│   ├── crds.yaml
+│   ├── deployments.yaml
+│   └── rbac.yaml
+└── metadata
+    ├── bundle.yaml
+    ├── annotations.yaml
+    └── dependencies.yaml
+
+2 directories, 6 files
+```
+
+Add a new command to `opm`, `opm bundle build`, which takes as input a directory tree -- described in [File Format](#file-format) -- and outputs an Operator Bundle image.
+
+_Ex._
 
-#### Required Manifests
+```sh
+$ opm bundle build . # detect the mediatype of the current directory, build it, and if successful, write the image tar.gz to stdout
+$ opm bundle build -o image://quay.io/my/bundle:v1.0.0 . # detect the mediatype of the current directory, build it, and if successful, push the image to a remote reference
+```
+#### Additively Build an Index
 
-The absence of a CSV means that there is a set of manifests that must ALWAYS be present to build `registry+v2` bundles. These manifests are ALWAYS __required__:
+As a Pipeline Maintainer, I want to additively build an index of `registry+v2` bundles so that I can release new bundles to an existing index.
 
-- a `Deployment` for the operator
-- a `metadata/olm.yaml` file with the following fields
-  - name
-  - version
-  - minKubeVersion
-  - installModes
+##### UX
 
-In certain situations, more manifests are required:
+Extend the `opm index add` command to accept `registry+v2` bundles.
 
-- providing an API from an APIService requires
-  - an `APIService`
-  - the `Service` it references
-- providing an admission webhook
-  - a `MutatingWebhookConfiguration` or a `ValidatingAdmissionWebhook`
-  - the `Service` it references
+#### Direct Application
```

Comment:
This is a good point. Without a CSV, how can a user install a `registry+v2` operator directly to the cluster? Basically, just apply a YAML stream directly via `opm` as you suggest. The end result is a CSV on the cluster, which may be a little unintuitive, but if we provide good docs on this should be fine.

Reply:
Users are used to this sort of indirection though. Creating a deployment on a cluster doesn't mean the user has to understand everything about defining a `ReplicatSet` or even the spec of their pod (though they do have the option to do that if they so choose)